apps/avifenc.c

```diff
@@ -2285,12 +2285,20 @@ MAIN()
         // image didn't provide any CICP. Explicitly signal SRGB CP/TC here, as 2/2/x will be
         // interpreted as SRGB anyway.
         image->colorPrimaries = AVIF_COLOR_PRIMARIES_SRGB;
+        image->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB;
+
 #if defined(AVIF_ENABLE_EXPERIMENTAL_JPEG_GAIN_MAP_CONVERSION)
-        if (image->gainMap && image->gainMap->image) {
-            image->gainMap->altColorPrimaries = AVIF_COLOR_PRIMARIES_SRGB;
+        if (image->gainMap && !image->gainMap->altICC.size) {
+            if (image->gainMap->altColorPrimaries == AVIF_COLOR_PRIMARIES_UNSPECIFIED) {
+                // Assume the alternate image has the same primaries as the base image.
+                image->gainMap->altColorPrimaries = image->colorPrimaries;
```

Comment:
[Note: This comment also applies to the other file.]

Because of line 2287 above, this is equivalent to setting `altColorPrimaries` to `AVIF_COLOR_PRIMARIES_SRGB`.

Line 2298 below sets `altTransferCharacteristics` to `AVIF_TRANSFER_CHARACTERISTICS_PQ` But PQ is usually used with color prrimaries 9 (`AVIF_COLOR_PRIMARIES_BT2020` or `AVIF_COLOR_PRIMARIES_BT2100`).

Should we set `altColorPrimaries` to `AVIF_COLOR_PRIMARIES_BT2100` here?

Or perhaps this should not be nested inside the if statement that starts at line 2282?

Reply:
> Should we set altColorPrimaries to AVIF_COLOR_PRIMARIES_BT2100 here?

No, if altColorPrimaries is not set we should assume that it's the same primaries as the base image. In practice right now the only way we can get a gain map is though a jpeg-r file, which does not allow specifying different color primaries for the alternate image, it's always the same as the base image. 
In avifjpeg.c we already set altColorPrimaries to the same value as the image
https://github.com/AOMediaCodec/libavif/blob/e0778af38864bba88f2c6ff2cd30fc7abf007500/apps/shared/avifjpeg.c#L1143
But the issue is that avifenc.c only sets the default value (sRGB) here, after we are done reading the image. When we go through the line above in avifjpeg.c, colorPrimaries is 2, so altColorPrimaries also ends up being 2.
So the idea is to propagate the new value to the altColorPrimaries field. This is why the if is nested, but I don't think it matters much i practice and perhaps it's more confusing this way so I changed it.